config.js

```diff
@@ -29,42 +31,44 @@ config.mqtt = {
     /**
      * Host where the MQTT Broker is located.
      */
-    host: 'localhost',
+    host: process.env.MQTT_HOST || 'localhost',
```

Comment:
config.js is though to be a simple description of default values por configuration properties. The env var overriding is done in the agent initialization code (for instance have a look to https://github.com/telefonicaid/iotagent-ul/tree/master/lib/configService.js#L39).

In fact, it seems that at least part of the env var you are proposing (i.e. MQTT_HOST) are already covered by some existing one (IOTA_MQTT_HOST).

Reply:
@fgalan - if `config.js` is supposed to be a simple descriptor, then maybe this needs to be added to the documentation somewhere - maybe at the head of this file?

I agree that delegating changes to the `lib/configService.js` is the right way to go as it 
would leave `config.js` easier to read.

There are multiple places conf _could_ be overwritten:

*  `lib/configService.js`
* `COPY` statement overriding default config.js
* `ENV` variables in Dockerfile, command line
* Attach a file volume with custom config.
* Add an entrypoint bash script to provide defaults.

There are undoubtedly others, it would be nice if  (by convention) this could be reduced to as few places to look as possible. 

Whatever convention is decided to should be summarized in the associated `README` and annotated in the relevant files.

Reply:
> @fgalan - if config.js is supposed to be a simple descriptor, then maybe this needs to be added to the documentation somewhere - maybe at the head of this file?

Documentation is already provided, see https://github.com/telefonicaid/iotagent-ul/blob/master/docs/installationguide.md#configuration

What probably could be an improvement is a link to the head of the file to that document

Reply:
With regards to the several configuration entrypoints, from the very begining (i.e. much before docker stuff was introduced in this repository) the approach was thought to work at two levels:

* Configuration file (default: `config.js` although it can be overriden in the command line, as far as I remember). This is the one typically used by the developer.
* Environment variables, which overrides the configuration file. This is the one typically used by packaging/deployment systems (eg. RPM).

(This is supposedly explained in https://github.com/telefonicaid/iotagent-ul/blob/master/docs/installationguide.md#configuration but if the description is not clear enough, improvements are welcome)

Then at some time docker came into play. From my point of view, docker role is clear as a "packaging/deployment system", i.e. a environement variable inyector, perfectly compatible with the two-levels approach described above.